fe/fe-core/src/main/java/com/starrocks/metric/ResourceGroupMetricMgr.java

```diff
@@ -128,18 +85,75 @@ private static LongCounterMetric createQeuryResourceGroupMetrics(Map cacheMap, S
         return (LongCounterMetric) cacheMap.get(resourceGroupName);
     }
 
-    private static void createQueryResourceGroupLatency(String resourceGroupName) {
-        List<GaugeMetricImpl> metricsList = new ArrayList<>();
-        for (String label : QUERY_LATENCY_LABLE) {
-            GaugeMetricImpl<Double> metrics =
-                    new GaugeMetricImpl<>(QUERY_RESOURCE_GROUP_LATENCY, Metric.MetricUnit.MILLISECONDS,
-                            label + " of resource group query latency");
-            metrics.addLabel(new MetricLabel("type", label));
-            metrics.addLabel(new MetricLabel("name", resourceGroupName));
-            metrics.setValue(0.0);
-            MetricRepo.addMetric(metrics);
-            metricsList.add(metrics);
+    private static String checkAndGetWorkGroupName(ConnectContext ctx) {
+        SessionVariable sessionVariable = ctx.getSessionVariable();
+        if (!sessionVariable.isEnableResourceGroup() || !sessionVariable.isEnablePipelineEngine()) {
+            return null;
+        }
+        WorkGroup workGroup = ctx.getWorkGroup();
+        return workGroup == null ? "default_wg" : workGroup.getName();
+    }
+
+    private static QueryResourceGroupLatencyMetrics createQueryResourceGroupLatencyMetrics(ConnectContext ctx) {
+        String resourceGroupName = checkAndGetWorkGroupName(ctx);
+        if (resourceGroupName == null || resourceGroupName.isEmpty()) {
+            return null;
+        }
+        if (!RESOURCE_GROUP_QUERY_LATENCY_MAP.containsKey(resourceGroupName)) {
+            synchronized (RESOURCE_GROUP_QUERY_LATENCY_MAP) {
+                if (!RESOURCE_GROUP_QUERY_LATENCY_MAP.containsKey(resourceGroupName)) {
```

Comment:
Why not replace it with`putIfAbsent` ? 
Why ConcurrentHashMap needs `synchronized` ?

Reply:
To ensure that each resource group metrics are registered only once.
`putIfAbsent` will new a lot of QueryResourceGroupLatencyMetrics instance.

Reply:
How about using `computeIfAbsent` to avoid using `synchronized` and creating `QueryResourceGroupLatencyMetrics` multiple times?

Reply:
yes, it might be better code style to use computeIfAbsent, containsKey is 3~5 times faster than computeIfAbsent, but both are fast and can be ignored.